src/appleseed-max-impl/appleseedmetalmtl/appleseedmetalmtl.cpp

```diff
@@ -0,0 +1,884 @@
+    material_params.insert("osl_surface", shader_group_name);
+
+    const std::string instance_name =
+        insert_texture_and_instance(
+            assembly,
+            m_alpha_texmap,
+            false,
+            asr::ParamArray(),
+            asr::ParamArray()
+                .insert("alpha_mode", "detect"));
+    if (!instance_name.empty())
+        material_params.insert("alpha_map", instance_name);
+    else material_params.insert("alpha_map", m_alpha / 100.0f);
+
+    return asr::OSLMaterialFactory().create(name, material_params);
+}
+
+asf::auto_release_ptr<asr::Material> AppleseedMetalMtl::create_builtin_material(
+    asr::Assembly&  assembly,
+    const char*     name)
+{
+    asr::ParamArray material_params;
+    std::string instance_name;
+    const bool use_max_procedural_maps = true;
+
+    //
+    // BRDF.
+    //
+
+    {
+        asr::ParamArray bsdf_params;
+        bsdf_params.insert("mdf", "ggx");
+
+        // Normal Reflectance.
+        instance_name = insert_texture_and_instance(assembly, m_facing_tint_color_texmap, use_max_procedural_maps);
+        if (!instance_name.empty())
+            bsdf_params.insert("normal_reflectance", instance_name);
+        else
+        {
+            const auto color_name = std::string(name) + "_bsdf_normal_reflectance_color";
+            insert_color(assembly, m_facing_tint_color, color_name.c_str());
+            bsdf_params.insert("normal_reflectance", color_name);
+        }
+
+        // Edge Tint.
+        instance_name = insert_texture_and_instance(assembly, m_edge_tint_color_texmap, use_max_procedural_maps);
+        if (!instance_name.empty())
+            bsdf_params.insert("edge_tint", instance_name);
+        else
+        {
+            const auto color_name = std::string(name) + "_bsdf_edge_tint";
+            insert_color(assembly, m_edge_tint_color, color_name.c_str());
+            bsdf_params.insert("edge_tint", color_name);
+        }
+
+        // Reflectance.
+        instance_name = insert_texture_and_instance(assembly, m_reflectance_texmap, use_max_procedural_maps);
+        if (!instance_name.empty())
+            bsdf_params.insert("reflectance_multiplier", instance_name);
+        else bsdf_params.insert("reflectance_multiplier", m_reflectance / 100.0f);
+
+        // Roughness.
+        instance_name = insert_texture_and_instance(assembly, m_roughness_texmap, use_max_procedural_maps);
+        if (!instance_name.empty())
+            bsdf_params.insert("roughness", instance_name);
+        else bsdf_params.insert("roughness", m_roughness / 100.0f);
+
+        // Anisotropic.
+        instance_name = insert_texture_and_instance(assembly, m_anisotropy_texmap, use_max_procedural_maps);
+        if (!instance_name.empty())
+            bsdf_params.insert("anisotropic", instance_name);
+        else bsdf_params.insert("anisotropic", m_anisotropy);
+
+        // BRDF.
+        const auto bsdf_name = std::string(name) + "_bsdf";
```

Comment:
It would make sense to use `_brdf` instead of `_bsdf`, here and for related entities (e.g. normal reflectance color).